src/parser.rs

```diff
@@ -116,38 +126,48 @@ fn headers(input: Input) -> IResult<Input, (File, File)> {
     Ok((input, (oldfile, newfile)))
 }
 
-fn header_line_content(input: Input) -> IResult<Input, File> {
+fn header_line_content(input: Input<'_>) -> IResult<Input<'_>, File> {
     let (input, filename) = filename(input)?;
     let (input, after) = opt(preceded(space0, file_metadata))(input)?;
 
-    Ok((input, File {
-        path: filename,
-        meta: after.and_then(|after| match after {
-            Cow::Borrowed("") => None,
-            _ => Some(
-                DateTime::parse_from_str(after.as_ref(), "%F %T%.f %z")
-                .or_else(|_| DateTime::parse_from_str(after.as_ref(), "%F %T %z"))
-                .ok()
-                .map_or_else(|| FileMetadata::Other(after), FileMetadata::DateTime)
-            ),
-        }),
-    }))
+    Ok((
+        input,
+        File {
+            path: filename,
+            meta: after.and_then(|after| match after {
+                Cow::Borrowed("") => None,
+                _ => Some(
+                    DateTime::parse_from_str(after.as_ref(), "%F %T%.f %z")
+                        .or_else(|_| DateTime::parse_from_str(after.as_ref(), "%F %T %z"))
+                        .ok()
+                        .map_or_else(|| FileMetadata::Other(after), FileMetadata::DateTime),
+                ),
+            }),
+        },
+    ))
 }
 
 // Hunks of the file differences
-fn chunks(input: Input) -> IResult<Input, Vec<Hunk>> {
+fn chunks(input: Input<'_>) -> IResult<Input<'_>, Vec<Hunk>> {
     many1(chunk)(input)
 }
 
-fn chunk(input: Input) -> IResult<Input, Hunk> {
+fn chunk(input: Input<'_>) -> IResult<Input<'_>, Hunk> {
     let (input, ranges) = chunk_header(input)?;
     let (input, lines) = many1(chunk_line)(input)?;
 
     let (old_range, new_range) = ranges;
-    Ok((input,  Hunk{old_range, new_range, lines}))
+    Ok((
+        input,
+        Hunk {
+            old_range,
+            new_range,
+            lines,
+        },
+    ))
 }
 
-fn chunk_header(input: Input) -> IResult<Input, (Range, Range)> {
+fn chunk_header(input: Input<'_>) -> IResult<Input<'_>, (Range, Range)> {
```

Comment:
might be more of a rust question than specific to this change: is there significance to naming the lifetime `'_` instead of something like `'a`?

(i need to brush up on lifetime naming conventions -- whatever is idiomatic here i'm good with!)

Reply:
No, it's just implicit. I can explicitly name these lifetimes (e.g. `fn chunk_header<'a>(input: Input<'a> -> IResult<Input<'a>.....` would be equivalent) if you care.

Reply:
Ah, slick! Just caught up on the `'_` rfc from... wow 2018. Into it. And it seems old enough not to raise a minimum-supported compiler concern, it's just me that is out of date 😅